draft-ietf-quic-transport.md

```diff
@@ -1607,22 +1606,30 @@ NOT subsequently advertise a smaller offset.  A sender may receive WINDOW_UPDATE
 frames out of order; a sender MUST therefore ignore any reductions in flow
 control credit.
 
+A receiver MUST close the connection with a
+QUIC_FLOW_CONTROL_RECEIVED_TOO_MUCH_DATA error ({{error-handling}}) if the
+peer violates the advertised stream or connection flow control windows.
+
 A sender MUST send BLOCKED frames to indicate it has data to write but is
 blocked by lack of connection or stream flow control credit.  BLOCKED frames are
 expected to be sent infrequently in common cases, but they are considered useful
 for debugging and monitoring purposes.
 
 A receiver advertises credit for a stream by sending a WINDOW_UPDATE frame with
-the StreamID set appropriately.  A receiver may simply use the current received
-offset to determine the flow control offset to be advertised.
+the StreamID set appropriately. A receiver may use the current offset of data
+consumed to determine the flow control offset to be advertised.
+A receiver MAY send copies of a WINDOW_UPDATE frame in multiple packets in order
+to make sure that the sender receives it even if one of the packets is lost. A
+sender MUST ignore duplicate WINDOW_UPDATE frames.
```

Comment:
I think that ACK and WINDOW_UPDATE should form a new class of frames, those that are cumulative.  These require that senders generate another frame if they are identified as being lost, but they don't require verbatim retransmission.

Looking at the other frames in the list there:

It doesn't make a lot of sense to retransmit PING as though it were something that needed repair (treat the ping as failed and maybe try again, but don't use automated processing for it for risk of having the overall time measured.  It's possible you might want to send multiple PING frames for redundancy within an RTT.

RST_STREAM clearly needs repair and repeating it verbatim is probably the only rational choice.  CONNECTION_CLOSE is probably the same, though GOAWAY might fall into the cumulative bucket as well.